File: Chapter_2/INSERTION_SORT.py
#INSERTION SORT implementation

def Insertion(A):
	for i in range(1,len(A)):
		key=A[i]
		j=i-1
		while(j >= 0 and A[j]>A[j+1]):
			temp=A[j]
			A[j]=A[j+1]
			A[j+1]=temp
			j-=1
		A[j+1]=key
if __name__ == '__main__':
	A=[5,4,3,2,1]
	print A
	Insertion(A)
	print A

File: Chapter_2/INSERTION_SORT.c
/*
Insertion sort implementation
It takes two arguments i.e array and its length
and sorts the array.
*/

#include <stdio.h>
void Insertion(int A[],int n){
	int key,temp,j;
	for(int i=1;i<n;i++){
		key=A[i];
		j=i-1;
		while(j >= 0 && A[j]>A[j+1]){
			temp=A[j];A[j]=A[j+1];A[j+1]=temp;
			j-=1;
		}
		A[j+1]=key;
	}
}
int main(){
	int A[5]={5,4,3,2,1};
	for(int i=0;i<5;i++) printf("%d ",A[i]);printf("\n");
	Insertion(A,5);
	for(int i=0;i<5;i++) printf("%d ",A[i]);printf("\n");
}